src/main/java/com/purbon/kafka/topology/serdes/TopicCustomDeserializer.java

```diff
@@ -154,11 +141,30 @@ public TopicImpl deserialize(JsonParser parser, DeserializationContext context)
 
     topic.setSchemas(schemas);
 
+    Map<String, String> metadata = getMap(rootNode.get("metadata"));
+    topic.setMetadata(metadata);
+
     LOGGER.debug(
         String.format("Topic %s with config %s has been created", topic.getName(), config));
     return topic;
   }
 
+  private Map<String, String> getMap(JsonNode jsonNode) {
```

Comment:
cosmetic, but may be name _getOptionalMap_ ? what do you think?

Reply:
I would then expect Optional<Map<String, String>>  to be the return type.

Reply:
it would imply most probably, so please disregard. It works as we have and is clean. Thanks.